### PR TITLE
#33 - Allow configuration of save cache path. [semver:minor]

### PR DESCRIPTION
**Pull request by gmemstr**
We can't currently pass through a list of values for this, so only a
single value will be accepted.

Addresses #33

### CODE REVIEW

File: src/commands/save-cache.yml [hunk 1]
```diff
@@ -4,9 +4,13 @@ parameters:
     description: "The cache key to use. The key is immutable."
     type: string
     default: "go-mod"
+  path:
+    description: "Path to cache."
+    type: string
+    # /home/circleci/go is the GOPATH in the cimg/go Docker image
+    default: "/home/circleci/go/pkg/mod"
```
Comment (KyleTryon) [inline src/commands/save-cache.yml:11]: `/home/circleci/` would only apply for CircleCI convenience images. What if we changed this to `~/` ?
I see it was originally set this way, wondering if anyone has complained yet.

Comment (gmemstr) [reply src/commands/save-cache.yml:11]: I don't think there should be any impact from that - interestingly, the `install` command doesn't set a GOPATH by default, although I haven't heard of any impact due to this.

Comment (KyleTryon) [reply src/commands/save-cache.yml:11]: @gmemstr Unless I am misunderstanding, the issue would be on any non-circleci image this path will not exist. The orb should not be dependant on assuming it will be in the cimg/go Docker image

File: src/commands/save-cache.yml [hunk 2]
```diff
@@ -4,9 +4,13 @@ parameters:
     description: "The cache key to use. The key is immutable."
     type: string
     default: "go-mod"
+  path:
+    description: "Path to cache."
+    type: string
+    # /home/circleci/go is the GOPATH in the cimg/go Docker image
+    default: "~/go/pkg/mod"
```
Comment (felicianotech) [inline src/commands/save-cache.yml:11]: Please take a look at the git history for this file.

This was changed AWAY from using `~` or `GOPATH` because CircleCI doesn't parse either when saving the cache.

The fact that the default PATH only works on CircleCI is fine because on a non-CircleCI image, this PATH doesn't exist at all. The user would need to create one from scratch, in which case they'd need to set it here.

@gmemstr, @KyleTryon

Comment (gmemstr) [reply src/commands/save-cache.yml]: Reverted! Good call. I did test to make sure `~` would be expanded when saving the cache, but that might not cover some edge cases.

**Comment by KyleTryon** [comment]
Added [semver:minor] to title for release
